data_key.go

```
@@ -3,39 +3,273 @@ package mmdbwriter
+
+	digest, err := h.hashMapContents(value)
+	if err != nil {
+		return 0, err
+	}
+	h.rememberProbation(dataHashCacheEntry{identity: identity, hash: digest, mapValue: value})
+	return digest, nil
+}
+
+func (h *dataHasher) hashMapContents(value mmdbtype.Map) (uint64, error) {
+	// Both accumulators are commutative so map iteration order cannot affect the
+	// result. The second accumulator makes cancelling sums less likely.
```

golangci-lint (2.12.2)

[error] `cancelling` is a misspelling of `canceling`

(misspell)